### PR TITLE
update kmesh_support.md

### LINKED ISSUE #1082: Installation guide needs update
**Task description**:
For [installation guide](https://kmesh.net/en/docs/setup/quickstart/), it would be helpful to include:
- supported kubernetes versions matrix
- supported istio versions matrix as control plan 
- requirements on kernel versions

We also need to rephrase categories at <https://github.com/kmesh-net/kmesh/blob/main/docs/kmesh_support.md>, to reflect the "Dual-Engine Mode" and "Kernel-Native Mode"

**Solution**:

**Who can join or take the task**:

The good first issue is intended for `first-time contributors` to get started on his/her contributor journey.

After a contributor has successfully completed 1-2 good first issue's,
they should be ready to move on to `help wanted` items, saving the remaining `good first issue` for other new contributors.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Describes the os that can be deployed Kmesh.
It also states that Kmesh can be deployed as long as the kernel version is greater than 5.10.
**Which issue(s) this PR fixes**:
Fixes #1082

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/kmesh_support.md [hunk 1]
```diff
@@ -1,32 +1,16 @@
...
 
-### Kmesh enhance kernel support all feature
+| Mode | OS Version | Kernel Version | Release Path |
+| :-------------: | :-------------: | :-------------: | :-------------: |
+| Kernel-Native Mode | openEuler-23.03 | 6.1.19 | https://repo.openeuler.org/openEuler-23.03/ |
+|                    | openEuler-22.03-LTS-SP1 |     5.10.0     | https://repo.openeuler.org/openEuler-22.03-LTS-SP1/ |
+|                    | Ubuntu Server 22.04.3 LTS |     5.15.0   |         https://releases.ubuntu.com/jammy/          |
+| Duel-Engine Mode | openEuler-23.03 | 6.1.19 | https://repo.openeuler.org/openEuler-23.03/ |
+|                    | openEuler-22.03-LTS-SP1 |     5.10.0     | https://repo.openeuler.org/openEuler-22.03-LTS-SP1/ |
+|                    | Ubuntu Server 22.04.3 LTS |     5.15.0   |         https://releases.ubuntu.com/jammy/          |
```
Comment (hzxuzhonghu) [inline docs/kmesh_support.md:14]: This is still  consuding, can you remove the OS version for dual engine

And i think kernel native mode, this is also not quire right

File: docs/kmesh_support.md [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
+Kmesh uses istiod as a control plane and therefore Kmesh has some dependencies on istio versions and kubernetes versions.
 
-| Mode | OS Version | Kernel Version | Release Path |
+Kmesh has two different modes, `Kernel-Native Mode` and `Duel-Engine Mode`. While there is no difference in the OS kernel version required for the two modes, the supported istio versions differ. Therefore we explain them separately.
+
+- **Kmesh Duel-Engine Mode:**
+
+| Version | Request Kernel Version | Supported Istio Version | Support Kubernetes Version |
+| :-------------: | :-------------: | :-------------: | :-------------: |
+| main | >5.10 | 1.22, 1.23, 1.24 | 1.26, 1.27, 1.28, 1.29, 1.30, 1.31 |
```
Comment (hzxuzhonghu) [inline docs/kmesh_support.md:13]: >=5.10

File: docs/kmesh_support.md [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
+| 0.5 | >5.10 | 1.22, 1.23 | 1.26, 1.27, 1.28, 1.29, 1.30 |
```
Comment (hzxuzhonghu) [inline docs/kmesh_support.md:14]: v0.5.x

File: docs/kmesh_support.md (hunk 2 above)
Comment (hzxuzhonghu) [reply docs/kmesh_support.md:13]: s/main/latest ? I am not sure how other project describe it

Comment (LiZhenCheng9527) [reply docs/kmesh_support.md:13]: istio ues `master` to describe it

File: docs/kmesh_support.md (hunk 3 above)
Comment (LiZhenCheng9527) [reply docs/kmesh_support.md]: Done

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**